types/gulp-rename/index.d.ts

```diff
@@ -19,6 +20,6 @@ declare namespace rename {
 }
 
 declare function rename(name: string): NodeJS.ReadWriteStream;
-declare function rename(callback: (path: rename.ParsedPath) => any): NodeJS.ReadWriteStream;
+declare function rename(callback: (path: rename.ParsedPath, file?: File) => any): NodeJS.ReadWriteStream;
```

Comment:
followup to @stof's point: `file` should probably not be optional.

It should not be optional if gulp-rename *always* provides it. It should be optional if gulp-rename *sometimes* fails to provide it. But this is not how libraries usually call callbacks.

If the typical callback needs to handle both cases as below, then it should be optional:

```ts
rename((path, file) => {
  if (file)
    console.log('got a file')
  else
   console.log('did not get a file, this is fine')
})
```

Reply:
it is always passed

Reply:
Then: 

```suggestion
declare function rename(callback: (path: rename.ParsedPath, file: File) => any): NodeJS.ReadWriteStream;
```